pyanaconda/bootloader.py

```diff
@@ -800,7 +800,8 @@ def set_boot_args(self, *args, **kwargs):
         if usr_device:
             dracut_devices.extend([usr_device])
 
-        netdevs = storage.devicetree.getDevicesByInstance(NetworkStorageDevice)
+        netdevs = (d for d in storage.devices if (getattr(d, "complete", True) and
+            isinstance(d, NetworkStorageDevice)))
```

Comment:
This should be a list comprehension because ``netdevs`` is used at least twice in the block below.